File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/anoncreds/ProverGetCredentialsTest.java
package org.hyperledger.indy.sdk.anoncreds;


import org.hyperledger.indy.sdk.wallet.WalletInvalidQueryException;
import org.json.JSONArray;
import org.json.JSONObject;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;

public class ProverGetCredentialsTest extends AnoncredsIntegrationTest {

	@Test
	public void testProverGetCredentialsWorksForEmptyFilter() throws Exception {

		JSONObject json = new JSONObject();
		String filter = json.toString();

		String credentials = Anoncreds.proverGetCredentials(wallet, filter).get();

		JSONArray credentialsArray = new JSONArray(credentials);

		assertEquals(3, credentialsArray.length());
	}

	@Test
	public void testProverGetCredentialsWorksForFilterByIssuer() throws Exception {

		JSONObject json = new JSONObject();
		String filter = json.put("issuer_did", issuerDid).toString();

		String credentials = Anoncreds.proverGetCredentials(wallet, filter).get();

		JSONArray credentialsArray = new JSONArray(credentials);

		assertEquals(2, credentialsArray.length());
	}

	@Test
	public void testProverGetCredentialsWorksForFilterBySchema() throws Exception {

		JSONObject json = new JSONObject();
		String filter = json.put("schema_id", gvtSchemaId).toString();

		String credentials = Anoncreds.proverGetCredentials(wallet, filter).get();

		JSONArray credentialsArray = new JSONArray(credentials);

		assertEquals(2, credentialsArray.length());
	}

	@Test
	public void testProverGetCredentialsWorksForFilterBySchemaName() throws Exception {

		JSONObject json = new JSONObject();
		String filter = json.put("schema_name", "gvt").toString();

		String credentials = Anoncreds.proverGetCredentials(wallet, filter).get();

		JSONArray credentialsArray = new JSONArray(credentials);

		assertEquals(2, credentialsArray.length());
	}

	@Test
	public void testProverGetCredentialsWorksForFilterByCredDefId() throws Exception {
		JSONObject json = new JSONObject();
		String filter = json.put("cred_def_id", issuer1gvtCredDefId).toString();
		String credentials = Anoncreds.proverGetCredentials(wallet, filter).get();

		JSONArray credentialsArray = new JSONArray(credentials);

		assertEquals(1, credentialsArray.length());
	}

	@Test
	public void testProverGetCredentialsWorksForEmptyResult() throws Exception {

		JSONObject json = new JSONObject();
		String filter = json.put("issuer_did", issuerDid + "a").toString();

		String credentials = Anoncreds.proverGetCredentials(wallet, filter).get();

		JSONArray credentialsArray = new JSONArray(credentials);

		assertEquals(0, credentialsArray.length());
	}

	@Test
	public void testProverGetCredentialsWorksForInvalidFilterJson() throws Exception {

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletInvalidQueryException.class));

		JSONObject json = new JSONObject();
		String filter = json.put("issuer_did", 1).toString();

		Anoncreds.proverGetCredentials(wallet, filter).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/AnonCredsTests/ProverGetCredentialsTest.cs
using Hyperledger.Indy.AnonCredsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AnonCredsTests
{
    [TestClass]
    public class ProverGetCredentialsTest : AnonCredsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestProverGetCredentialsWorksForEmptyFilter()
        {
            var credentials = await AnonCreds.ProverGetCredentialsAsync(wallet, "{}");

            var credentialsArray = JArray.Parse(credentials);

            Assert.AreEqual(3, credentialsArray.Count);
        }

        [TestMethod]
        public async Task TestProverGetCredentialsWorksForFilterByIssuer()
        {
            var filter = string.Format("{{\"issuer_did\":\"{0}\"}}", issuerDid);

            var credentials = await AnonCreds.ProverGetCredentialsAsync(wallet, filter);

            var credentialsArray = JArray.Parse(credentials);

            Assert.AreEqual(2, credentialsArray.Count);
        }

        [TestMethod]
        public async Task TestProverGetCredentialsWorksForFilterBySchema()
        {
            var filter = string.Format("{{\"schema_id\":\"{0}\"}}", gvtSchemaId);

            var credentials = await AnonCreds.ProverGetCredentialsAsync(wallet, filter);

            var credentialsArray = JArray.Parse(credentials);

            Assert.AreEqual(2, credentialsArray.Count);
        }

        [TestMethod]
        public async Task TestProverGetCredentialsWorksForFilterBySchemaName()
        {
            var filter = "{\"schema_name\":\"gvt\"}";

            var credentials = await AnonCreds.ProverGetCredentialsAsync(wallet, filter);

            var credentialsArray = JArray.Parse(credentials);

            Assert.AreEqual(2, credentialsArray.Count);
        }


        [TestMethod]
        public async Task TestProverGetCredentialsWorksForFilterByCredDefId()
        {
            var filter = string.Format("{{\"cred_def_id\":\"{0}\"}}", issuer1gvtCredDefId);

            var credentials = await AnonCreds.ProverGetCredentialsAsync(wallet, filter);

            var credentialsArray = JArray.Parse(credentials);

            Assert.AreEqual(1, credentialsArray.Count);
        }

        [TestMethod]
        public async Task TestProverGetCredentialsWorksForEmptyResult()
        {
            var filter = string.Format("{{\"issuer_id\":\"{0}a\"}}",  issuerDid);

            var credentials = await AnonCreds.ProverGetCredentialsAsync(wallet, filter);

            var credentialsArray = JArray.Parse(credentials);

            Assert.AreEqual(0, credentialsArray.Count);
        }


        [TestMethod]
        public async Task TestProverGetCredentialsWorksForInvalidFilterJson()
        {
            var filter = "{\"issuer_id\":1}";

            var ex = await Assert.ThrowsExceptionAsync<WalletInvalidQueryException>(() =>
                AnonCreds.ProverGetCredentialsAsync(wallet, filter)
            );
        }

    }
}
